### ISSUE TITLE
OPM Map support broken

### ISSUE THREAD

**Issue by dakoop** [VisTrails/VisTrails#1076]
Trying to export a workflow with a `Map` module to OPM leads to a traceback. The example workflow was:

ConcatenateString(str1="a") --(self,functionPort)-> Map(InputList=['a','b','c'],InputPort=['str2'],OutputPort='value') --(Result, value)-> StandardOutput

Run this workflow and then try to export to OPM. The following traceback should be generated:
```
Traceback (most recent call last):
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/gui/vistrails_window.py", line 1937, in method
    getattr(obj, method_name)()
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/gui/vistrail_view.py", line 1043, in save_opm
    self.controller.write_opm(locator)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/gui/vistrail_controller.py", line 1119, in write_opm
    locator.save_as(opm_graph)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/core/db/locator.py", line 158, in save_as
    obj = _XMLFileLocator.save(self, obj, True, version)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/locator.py", line 511, in save
    obj = io.save_to_xml(obj, self._name, version)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/io.py", line 569, in save_to_xml
    return save_opm_to_xml(obj, filename, version)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/io.py", line 1351, in save_opm_to_xml
    opm_graph.registry)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/opm.py", line 665, in create_opm
    process_workflow(workflow, log, account, {}, {}, 0, True) 
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/opm.py", line 649, in process_workflow
    upstream_artifacts, downstream_artifacts)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/opm.py", line 587, in process_exec
    special_ports[2](module, found_input_ports, found_output_ports)
  File "VisTrails 2.2.app/Contents/Resources/lib/python2.7/vistrails/db/services/opm.py", line 305, in process_map
    if loop_exec.db_completed == 1:
AttributeError: 'LoopExec' object has no attribute 'db_completed'
```

**Comment by remram44** [comment]
This is because of the changes on [`logger-rewrite`](https://github.com/VisTrails/VisTrails/commit/7216d0a52db8e521597a5fa9648c6963be0fd27b) ([diff](https://github.com/VisTrails/VisTrails/compare/27d7aa484bb0d71e521374ce32a6797aa1660664...7216d0a52db8e521597a5fa9648c6963be0fd27b)) I assume.

`completed` is no longer on loop_exec, it's on loop_iteration.

**Comment by rexissimus** [comment]
It also looks like anything created with the "Control Flow assistant" breaks, because `InputPort` is assumed to be a parameter, not a connection!

Handling the control-flow modules is complicated, because it has to, e.g., compute which branch an `If` module executes. We should probably have more annotations in the execution provenance to make this easier.

Provenance for the automatic looping is easier to compute, but probably needs some work to produce more complete provenance.

**Comment by rexissimus** [comment]
Fixed by 8eb572d6d727a33f7172381d264cf632cd1ce5db.
